pallets/cash/src/pipeline.rs

```diff
@@ -247,59 +268,40 @@ impl State {
         if let Some(total_cash_principal_new) = self.total_cash_principal {
             TotalCashPrincipal::put(total_cash_principal_new);
         }
+        self.chain_cash_principals
+            .iter()
+            .for_each(|(chain_id, chain_cash_principal)| {
+                ChainCashPrincipals::insert(chain_id, chain_cash_principal);
+            });
     }
 }
 
-fn prepare_transfer_asset<T: Config>(
+fn prepare_augment_asset<T: Config>(
```

Comment:
transfer is now a `augment` for the recipient followed by a `reduce` for the sender. It makes it excessively clean to implement lock and extract then (since they are _just_ one half of transfer, each)